nevergrad/benchmark/experiments.py

```diff
@@ -1181,5 +1217,45 @@ def bragg_structure(seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
         for algo in default_optims if default_optims is not None else optims:
             yield Experiment(func, algo, int(budget), num_workers=1, seed=xpseed)
         for f in [func, func_nostruct, func_mix]:
-            for algo2 in recombinable:
-                yield Experiment(f, algo2, int(budget), num_workers=1, seed=xpseed)
+            for algo in recombinable:
+                yield Experiment(f, algo, int(budget), num_workers=1, seed=xpseed)
+
+
+# TODO : complete here for adversarial attacks.
+@registry.register
+def AdversarialAttack(seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
+    class Normalize(nn.Module):
+        def __init__(self, mean, std):
+            super().__init__()
+            self.mean = torch.Tensor(mean)
+            self.std = torch.Tensor(std)
+
+        def forward(self, x):
+            return (x - self.mean.type_as(x)[None, :, None, None]) / self.std.type_as(x)[None, :, None, None]
+
+    class Resnet50(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.norm = Normalize(mean=[0.485, 0.456, 0.406],
+                                  std=[0.229, 0.224, 0.225])
+            self.model = resnet50(pretrained=True)
+
+        def forward(self, x):
+            return self.model(self.norm(x))
+
+    image_size = 224
+    data_folder = "/datasets01_101/imagenet_full_size/061417/val"
+    data_loader = torch.utils.data.DataLoader(
+        torchvision.datasets.ImageFolder(data_folder,
+                                         transforms.Compose([
+                                             transforms.Resize(image_size),
+                                             transforms.CenterCrop(image_size),
+                                             transforms.ToTensor()])),
+        batch_size=1,
+        shuffle=True,
+        num_workers=8,
+        pin_memory=True)
+
+    classifier = Resnet50()
+
+    # TODO: Add experiment here
```

Comment:
The end of the function is just something like "optims = ["DE", "PSO", "DiagonalCMA", "CMA"] and the kind of loop you can see in other xps.

Reply:
yes but I do not know exactly what  to put inside, let see on Wed